src/Shared/SqlProcessor.cs

```diff
@@ -87,62 +456,111 @@ private static bool SkipComment(string sql, ref int index)
                 }
             }
 
-            index = i;
+            state.ParsePosition = ++i;
             return true;
+#endif
         }
 
         // Scan past single-line comment
         if (ch == '-' && i + 1 < length && sql[i + 1] == '-')
         {
+#if NET
+            // Find next line break efficiently and preserve the newline for whitespace handling
+            var rest = sql.Slice(i + 2);
+            var idx = rest.IndexOfAny('\r', '\n');
+            if (idx >= 0)
+            {
+                // Position at the newline so ParseWhitespace can copy it
+                state.ParsePosition = i + 2 + idx;
+            }
+            else
+            {
+                state.ParsePosition = sql.Length;
+            }
+
+            return true;
+#else
             for (i += 2; i < length; ++i)
             {
                 ch = sql[i];
-                if (ch is '\r' or '\n')
+                if (ch == '\r' || ch == '\n')
```

Comment:
Is this faster than `is` and `or`?

Reply:
I don't believe so, but I haven't tested. I think they should end up compiling to the same IL.